Source/WebKit/WPEPlatform/wpe/wayland/WPEViewWayland.cpp

```diff
@@ -406,6 +425,11 @@ static const struct zwp_linux_dmabuf_feedback_v1_listener linuxDMABufFeedbackLis
     [](void* data, struct zwp_linux_dmabuf_feedback_v1*)
     {
         auto* view = WPE_VIEW_WAYLAND(data);
+
+        // Compositor might not have sent the format table. In that case, try to reuse the previous one.
+        if (!view->priv->pendingDMABufFeedback->formatTable && view->priv->committedDMABufFeedback && view->priv->committedDMABufFeedback->formatTable)
+            view->priv->pendingDMABufFeedback->formatTable = WTFMove(view->priv->committedDMABufFeedback->formatTable);
+
```

Comment:
We need to handle here the case of format table being invalid and set pending feedback to nullptr in that case. Because in current code we know that if we have a committed feedback, it has a valid table.

Reply:
Right. If we end up without any usable formats table at the `done` callback (either because one wasn't sent, or we don't have a previously committed one) then we need to bail out early and avoid sending the signal as well. Good catch!